subprojects/platform-native/src/docs/sample-templates/source-globbing/README.adoc

```diff
@@ -0,0 +1,165 @@
+:an-application: a library
+:c: Swift
+:source-file-extension: swift
+:application: library
+endif::[]
+
+:ref-plugin: <<plugin:{plugin-name}, {c-application} plugin>>
+
+ifndef::c[ERROR: Please define `plugin-name` attribute.]
+
+= {c-application} Custom Source Layout Globbing
+:summary: Set custom source layout Globbing for a {c-application}.
+:type: sample-chapter
+:tags: sample, {application}, sources, native, {C}, gradle
+:category: {C}
+:description: See how to set the source layout globbing of {an-application} implemented in {C} using the Gradle Nokee plugins.
+:ref-configurable-file-tree: https://docs.gradle.org/current/javadoc/org/gradle/api/file/ConfigurableFileTree.html
+:ref-configurable-file-tree-exclude: https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/util/PatternFilterable.html#exclude-java.lang.String...-
+:ref-configurable-file-tree-matching: https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileTree.html#matching-groovy.lang.Closure-
+
+NOTE: You can open this sample inside an IDE using the https://www.jetbrains.com/help/idea/gradle.html#gradle_import_project_start[IntelliJ native importer] or https://projects.eclipse.org/projects/tools.buildship[Eclipse Buildship].
+
+Here is a sample where we set the source layout globbing of {an-application} implemented in {C} with no dependencies and minimal configuration.
+
+====
+[.multi-language-sample]
+=====
+.build.gradle
+[source,groovy]
+----
+include::groovy-dsl/build.{plugin-name}.gradle[]
+----
+=====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+include::kotlin-dsl/build.{plugin-name}.gradle.kts[]
+----
+=====
+====
+<1> Apply the {ref-plugin}.
+<2> Set {C} sources to *include* all files with `{source-file-extension}` extension recursively from the `srcs` directory.
+ifdef::header-file-extension[]
+<3> Set private headers to *include* all files with `{header-file-extension}` extension recursively from the `hdrs` directory.
+ifeval::["{entry-point}" == "library"]
+<4> Set public headers to *include* all files with `{header-file-extension}` extension recursively from the `incs` directory.
+endif::[]
+endif::[]
+
+
+NOTE: More filter patterns exist such as link:{ref-configurable-file-tree-exclude}[*exclude*], link:{ref-configurable-file-tree-matching}[*matching*], etc.
+Refer to the link:{ref-configurable-file-tree}[Gradle documentation for more information].
+
+Demonstration below where we see the {application} builds:
+
+[source,terminal]
+----
+$ ./gradlew assemble
+
+BUILD SUCCESSFUL
+4 actionable tasks: 4 executed
+----
+
+Demonstration when including a file that doesn't meet the source filters:
+
+[source,terminal]
+----
+$ echo "broken!" > srcs/a_broken.potato     # <1>
+$ ./gradlew clean assemble                  # <2>
+
+BUILD SUCCESSFUL                            # <3>
+4 actionable tasks: 4 executed
+----
+<1> Create a broken {C} source file in conventional location with no `{source-file-extension}` extension.
```

Comment:
❌ I'm unsure how we should reword this, but the part "conventional location with no C extension" is not exactly correct here. We place the file `a_broken.potato` in the directory (e.g. `srcs`) that will be globbed using the pattern `**/*.c` which means any files that have a different file extension than `.c` won't be picked up. In this case, the file extension is `.potato` meaning it will be ignored.

Reply:
Maybe, reverse the logic to use `srcs/a_broken.c` and show the build fails because `.c` file is taken into account.

Important to note that:
- From all the files under `srcs` only `**/*.c` are considered
- All other files are ignored